pkg/build/api/validation/validation.go

```diff
@@ -176,6 +180,17 @@ func validateStrategy(strategy *buildapi.BuildStrategy) errs.ValidationErrorList
 	return allErrs
 }
 
+func validateDockerSecretRef(ref *kapi.ObjectReference) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if ref == nil {
+		return allErrs
+	}
+	if len(ref.Name) == 0 {
+		allErrs = append(allErrs, errs.NewFieldInvalid("RegistrySecret", ref.Name, "the Secret name cannot be blank"))
+	}
+	return allErrs
```

Comment:
If I do, then users will have to add Kind to ObjectReference... Is that
what you mean?
On Mar 23, 2015 7:34 PM, "Jordan Liggitt" <notifications@github.com> wrote:

> In pkg/build/api/validation/validation.go
> <https://github.com/openshift/origin/pull/1411#discussion_r26965430>:
>
> > @@ -176,6 +180,17 @@ func validateStrategy(strategy *buildapi.BuildStrategy) errs.ValidationErrorList
> >  	return allErrs
> >  }
> >
> > +func validateDockerSecretRef(ref *kapi.ObjectReference) errs.ValidationErrorList {
> > +	allErrs := errs.ValidationErrorList{}
> > +	if ref == nil {
> > +		return allErrs
> > +	}
> > +	if len(ref.Name) == 0 {
> > +		allErrs = append(allErrs, errs.NewFieldInvalid("RegistrySecret", ref.Name, "the Secret name cannot be blank"))
> > +	}
> > +	return allErrs
>
> Make sure the kind is what you expect. Not sure what to check for the
> namespace
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/1411/files#r26965430>.
>